src/app/FakeLib/CMakeHelper.fs

```diff
@@ -0,0 +1,286 @@
+    // Try to find the CMake executable if not specified by the user.
+    let cmakeExe =
+        if not <| isNullOrEmpty toolPath then toolPath else
+        let found = findCMakeExe "cmake"
+        if found <> None then found.Value else failwith "Cannot find the CMake executable."
+    let setInfo (info:ProcessStartInfo) =
+        info.FileName <- cmakeExe
+        info.WorkingDirectory <- binaryDir
+        info.Arguments <-
+            // CMake expects the binary directory to be passed as an argument.
+            if (String.IsNullOrEmpty args) then "\"" + binaryDir + "\"" else args
+    let result = ExecProcess (setInfo) (TimeSpan.FromMinutes 5.0)
+    if result <> 0 then failwithf "CMake failed with exit code %i." result
+
+/// Calls `cmake` to generate a project.
+/// ## Parameters
+///  - `setParams` - Function used to manipulate the default CMake parameters. See `CMakeGenerateParams`.
+let CMakeGenerate setParams =
+    let parameters = setParams CMakeGenerateDefaults
+    // CMake expects an existing source directory.
+    // Not defaulted because it would prevent building multiple CMake projects in the same FAKE script.
+    if isNullOrEmpty parameters.SourceDirectory then failwith "The CMake source directory is not set."
+    let argsIfNotEmpty format values =
+        List.filter (isNotNullOrEmpty) values
+        |> List.map (sprintf format)
+    let generator = argsIfNotEmpty "-G \"%s\"" [parameters.Generator]
+    let toolchain = argsIfNotEmpty "-D CMAKE_TOOLCHAIN_FILE:FILEPATH=\"%s\"" [FormatCMakePath parameters.Toolchain]
+    let toolset = argsIfNotEmpty "-T \"%s\"" [parameters.Toolset]
+    let platform = argsIfNotEmpty "-A \"%s\"" [parameters.Platform]
+    let caches = parameters.Caches |> List.map FormatCMakePath |> argsIfNotEmpty "-C \"%s\""
+    let variables =
+        parameters.Variables
+        |> List.map (fun option ->
+            "-D " + option.Name +
+            match option.Value with
+            | CMakeBoolean(value) -> ":BOOL=" + if value then "ON" else "OFF"
+            | CMakeString(value) -> ":STRING=\"" + value + "\""
+            | CMakeDirPath(value) -> FormatCMakePath value |> sprintf ":PATH=\"%s\""
+            | CMakeFilePath(value) -> FormatCMakePath value |> sprintf ":FILEPATH=\"%s\""
+        )
+    let cacheEntriesToRemove = argsIfNotEmpty "-U \"%s\"" parameters.CacheEntriesToRemove
+    let args =
+        [ generator; toolchain; toolset; platform; caches; variables; cacheEntriesToRemove
+          [parameters.AdditionalArgs; "\"" + parameters.SourceDirectory + "\""]
+        ] |> List.concat |> String.concat " "
+    callCMake parameters.ToolPath parameters.BinaryDirectory args parameters.Timeout
+
+/// Calls `cmake --build` to build a project.
+/// ## Parameters
+///  - `setParams` - Function used to manipulate the default CMake parameters. See `CMakeBuildParams`.
+let CMakeBuild setParams =
+    let parameters = setParams CMakeBuildDefaults
```

Comment:
OK.

Is there a need for tracing the whole `generate` and `build` tasks or would using them from `callCMake` be sufficient (it would simplify the build of the description string)?

More globally, what is the point of these functions?